engines/scumm/script_v6.cpp

```diff
@@ -495,6 +495,80 @@ void ScummEngine_v6::o6_pushByteVar() {
+					if (sprintCounter >= 11) {
+						if (vm.localvar[_currentScript][5] == 12 || vm.localvar[_currentScript][5] == 11 || vm.localvar[_currentScript][5] == 10) {
+							sprintGain = 2;
+						} else if (vm.localvar[_currentScript][5] == 9) {
+							sprintGain = 3;
+						} else {
+							sprintGain = 4;
+						}
+					} else if (sprintCounter >= 9) {
+						if (vm.localvar[_currentScript][5] == 12 || vm.localvar[_currentScript][5] == 11 || vm.localvar[_currentScript][5] == 10) {
+							sprintGain = 2;
+						} else {
+							sprintGain = 3;
+						}
+					} else if (sprintCounter >= 7) {
+							sprintGain = 2;
+					} else {
+						sprintGain = 1;
+					}
+					writeVar(0x4000 + 4, sprintGain);
+				}
+			}
+
+			// This code will change the velocity of the hit based on the pitch thrown, and the location of the pitch itself.
+			if (_game.id == GID_BASEBALL2001 && _currentRoom == 4 && vm.slot[_currentScript].number == 2090 && readVar(399) == 1) {
+				int offset = _scriptPointer - _scriptOrgPointer;
+				int powerAdjustment = vm.localvar[_currentScript][4];
+
+				// Checks if the swing is either Power or Line Drive
+				if (offset == 102789 && (readVar(387) == 1||readVar(387) == 2)) {
+					// Checks if the current pitch type is the same as that of the "remembered" pitch type
+					if (readArray(346, 0, 0) == readArray(346, 1, 0)) {
+						// Checks if the current pitch is either a Heat or a Fireball. The reason it adds less than the other pitches
+						// is because in the actual calculation it adds 5 to these two anyway, so this should also balance them out.
+						if (readVar(0x8000 + 10) == 14 || readVar(0x8000 + 10) == 21) {
+							powerAdjustment = powerAdjustment + 15;
+						} else {
+							powerAdjustment = powerAdjustment + 20;
+						}
+					}
+					// Checks if the zone location is the same as that of the previous one. This should slightly reduce the amount of pitching to the exact same location.
+					// Can also be adjusted later if necessary.
+					if (readArray(346, 0, 1) == readArray(346, 1, 1)) {
+						powerAdjustment = powerAdjustment + 20;
+					}
+
+					// write the power adjustment to the result
+					writeVar(0x4000 + 4, powerAdjustment);
+				}
+			}
+
+			// Remember the previous pitch thrown and the previous pitch "zone location", then set those two values to the "remembered" values for later use.
+			if (_game.id == GID_BASEBALL2001 && _currentRoom == 4 && vm.slot[_currentScript].number == 2201 && readVar(399) == 1) {
+				writeArray(346, 1, 0, readArray(346, 0, 0));
+				writeArray(346, 1, 1, readArray(346, 0, 1));
+			}
+
+			// This sets the base cost of a slow ball to 2. Previously it costed the least of every pitch to throw, which resulted in people only using that pitch.
+			if (_game.id == GID_BASEBALL2001 && _currentRoom == 4 && vm.slot[_currentScript].number == 2057 && readVar(399) == 1) {
+				if (readVar(0x4000 + 1) == 15) {
+					writeVar(0x4000 + 2, 2);
+				}
+			}
+		}
+	#endif
```

Comment:
Same as before, we prefer macros to be placed at the beginning of a line, without preceding tabs or spaces.

Reply:
Sounds good! I will look into all of these.

Reply:
Let me know if I got it right. I misread it at first haha